site/book/06-deploying-packages/01-initializing-a-package-for-apply.md

```diff
@@ -0,0 +1,36 @@
+Before you can apply the package to the cluster, it needs to be initialized
+using `live init`. This is a one-time client-side operation that adds metadata
+to the `Kptfile` specifying a `ResourceGroup` resource on the cluster associated
+with this package. The `ResourceGroup`resource will be created by the
+`live apply` command.
+
+Let's initialize the `wordpress` package:
+
+```shell
+$ kpt live init wordpress
+initializing Kptfile inventory info (namespace: default)...success
+```
+
+This adds the `inventory` section to the `Kptfile`:
+
+```yaml
+# wordpress/Kptfile (Excerpt)
+inventory:
+  namespace: default
+  name: inventory-10285025
+  inventoryID: 06ca0268f3ccda82bfd44bec273530a4f72fa9a7-1623278424452000000
+```
+
+`ResourceGroup` is a namespace-scoped resource. By default, `live init` command
+uses heuristics to automatically choose the namespace. In this example, all the
+resources in `wordpress` package were in the `default` namespace, so it chose
+`default` for the namespace. Alternatively, you can manually configure the name
+and namespace of the `ResourceGroup` resource.
+
+?> Refer to the [init command reference][init-doc] for usage.
+
+!> Once a package is applied to the cluster, you do not want to change the
+metadata in the `inventory` section as it severs the association between the
+package and the `ResourceGroup` leading to destructive operations.
```

Comment:
I am thinking about a common CUJ:

I have a package A which I tested by `kpt live init` and then applying to my cluster. Everything looked good, so I uprevved (or tagged) the package A. So inventory section is in the Kptfile. 

Downstream consumer of package A (let's call it A') has already applied the package in their cluster, so they already have an inventory section in the Kptfile of A'. And, now they run `pkg update A'`

What is going to happen here ? Will `pkg update A'` going to ignore the `inventory` section from the package A ?
Ignoring the `inventory` section from upstream seems reasonable ? right ?

/cc @mortent 

Or may be, we need to add a section in the book about `publishing a package` where we can talk about the best practices about publishing a package that can eventually be codified in the `kpt pkg publish` command.

Reply:
Publishing a package with the inventory information would not be recommended. The inventory is specific to applying the packages in a specific cluster, so adding this information to the Kptfile should probably happen once the package has been customized. This will somewhat depend on how users want to handle things like creating variants of a package for different environments.

We have talked about adding a separate publish command. A first step might be to add a command that can provide the validation of the package prior to publishing. Some things that could be included are:
 * No inventory information.
 * Package is hydrated, i.e. running the `kpt fn render` command should not lead to a diff.
 * The `UpstreamLock` information matches the `Upstream` information in all subpackages.
 * ...

Reply:
I think we will need a publish command, otherwise, there's a growing laundry list of things the publisher needs to manually watch out for.